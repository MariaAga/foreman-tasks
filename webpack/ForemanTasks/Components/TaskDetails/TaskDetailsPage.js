import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import PageLayout from 'foremanReact/pages/common/PageLayout/PageLayout';
import TaskDetails from './TaskDetails';
import { getTaskID } from './TasksDetailsHelper';

export const indexBreadcrumbItem = {
  caption: __('Tasks'),
  url: '/foreman_tasks/tasks',
};

export const breadcrumbTaskItem = task => ({
  caption: task.action || __('N/A'),
  url: `/foreman_tasks/tasks/${getTaskID()}/details`,
});

class TaskDetailsPage extends Component {
  componentDidMount() {
    const { timeoutId, refetchTaskDetails, fetchTaskDetails } = this.props;

    fetchTaskDetails(getTaskID(), timeoutId, refetchTaskDetails);
  }
  componentWillUnmount() {
    this.props.taskReloadStop(this.props.timeoutId);
  }
  render() {
    return (
      <PageLayout
        searchable={false}
        breadcrumbOptions={{
          isSwitchable: true,
          breadcrumbItems: [
            indexBreadcrumbItem,
            breadcrumbTaskItem(this.props),
          ],
          resource: {
            nameField: 'action',
            resourceUrl: '/foreman_tasks/api/tasks',
            switcherItemUrl: '/foreman_tasks/tasks/:id',
          },
        }}
      >
        <TaskDetails {...this.props} />
      </PageLayout>
    );
  }
}

TaskDetailsPage.propTypes = {
  fetchTaskDetails: PropTypes.func,
  refetchTaskDetails: PropTypes.func,
  taskReloadStop: PropTypes.func,
  timeoutId: PropTypes.number,
};

TaskDetailsPage.defaultProps = {
  fetchTaskDetails: () => null,
  refetchTaskDetails: () => null,
  taskReloadStop: () => null,
  timeoutId: null,
};
export default TaskDetailsPage;
