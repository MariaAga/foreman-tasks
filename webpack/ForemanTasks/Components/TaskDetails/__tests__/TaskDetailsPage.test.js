import { testComponentSnapshotsWithFixtures } from 'react-redux-test-utils';

import TaskDetailsPage from '../TaskDetailsPage';

const fixtures = {
  'render without Props': {},
};

describe('TaskDetailsPage', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TaskDetailsPage, fixtures));
});
