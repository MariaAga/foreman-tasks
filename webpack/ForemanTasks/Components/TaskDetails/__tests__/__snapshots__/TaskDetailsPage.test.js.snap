// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskDetailsPage rendering render without Props 1`] = `
<PageLayout
  breadcrumbOptions={
    Object {
      "breadcrumbItems": Array [
        Object {
          "caption": "Tasks",
          "url": "/foreman_tasks/tasks",
        },
        Object {
          "caption": "N/A",
          "url": "/foreman_tasks/tasks//details",
        },
      ],
      "isSwitchable": true,
      "resource": Object {
        "nameField": "action",
        "resourceUrl": "/foreman_tasks/api/tasks",
        "switcherItemUrl": "/foreman_tasks/tasks/:id",
      },
    }
  }
  searchable={false}
>
  <TaskDetails
    action=""
    allowDangerousActions={false}
    cancellable={false}
    endedAt=""
    error={Array []}
    executionPlan={Object {}}
    externalId=""
    failedSteps={Array []}
    fetchTaskDetails={[Function]}
    hasSubTasks={false}
    help=""
    id=""
    input={Array []}
    label=""
    locks={Array []}
    output={Object {}}
    parentTask=""
    progress={0}
    refetchTaskDetails={[Function]}
    result="error"
    resumable={false}
    runningSteps={Array []}
    showForceUnlockModal={false}
    showUnlockModal={false}
    startAt=""
    startBefore=""
    startedAt=""
    state=""
    taskReload={false}
    taskReloadStart={[Function]}
    taskReloadStop={[Function]}
    timeoutId={null}
    toggleForceUnlockModal={[Function]}
    toggleUnlockModal={[Function]}
    username=""
    usernamePath=""
  />
</PageLayout>
`;
